bundle of doc improvements

robstoll opened the pull request.

robstoll commented on docs/docs/reference/metaprogramming/tasty-reflect.md:
````diff
@@ -117,17 +116,17 @@ def macroImpl()(qctx: QuoteContext): Expr[Unit] = {
 
 ```scala
-def collectPatternVariables(tree: Tree)(implicit ctx: Context): List[Symbol] = {
+def collectPatternVariables(tree: Tree)(implicit qctx: QuoteContext): List[Symbol] = {
````
please check this one carefully, it was more a guess that it should be QuoteContext

nicolasstucki replied:
Actually it is `Context` (the equivalent to a `qctx.tasty.Context`)

robstoll replied:
reverted.

odersky commented on docs/docs/reference/changed-features/implicit-resolution.md:
```diff
@@ -7,39 +7,39 @@ Implicit resolution uses a new algorithm which caches implicit results
 affect implicits on the language level.
 
- 1. Types of implicit values and result types of implicit methods
+ 1. Types of `given` values and result types of `given` methods
```
No, here we have to keep `implicit`. These changes apply to old-style implicit vals, not new givens. So most changes in this file need to be reverted. I'll only mark those we can keep.

odersky commented on docs/docs/reference/changed-features/implicit-resolution.md:
```diff
@@ -50,9 +50,9 @@ affect implicits on the language level.
+    Both `a` and `b` are visible as givens at the point of the definition
     of `type C`. However, a reference to `p.o.C` outside of package `p` will
-    have only `b` in its implicit scope but not `a`.
+    have only `b` in its implicit search scope but not `a`.
```
this change is OK

odersky commented on docs/docs/reference/changed-features/implicit-resolution.md:
```diff
@@ -7,39 +7,39 @@ Implicit resolution uses a new algorithm which caches implicit results
     no longer applies.
 
- 3. Package prefixes no longer contribute to the implicit scope of a type.
+ 3. Package prefixes no longer contribute to the implicit search scope of a type.
```
this change is OK

odersky commented on docs/docs/reference/changed-features/implicit-resolution.md:
```diff
@@ -82,23 +82,24 @@ affect implicits on the language level.
+ 5. The treatment of divergence errors has also changed. A divergent `given` is
     treated as a normal failure, after which alternatives are still tried. This also makes
-    sense: Encountering a divergent implicit means that we assume that no finite
-    solution can be found on the given path, but another path can still be tried. By contrast
```
given -> corresponding is OK

odersky commented on docs/docs/reference/new-types/intersection-types.md:
````diff
@@ -11,18 +11,18 @@ The type `S & T` represents values that are of the type `S` and `T` at the same
 
 ```scala
 trait Resettable {
-  def reset(): this.type
````
It needs a result type for sure. If not `this.type` then `Unit`.

robstoll replied:
sorry, was thinking in the wrong language. I thought absent type in trait = Unit. I'am going to use Unit I thought this.type could confuse beginners

odersky commented on docs/docs/reference/new-types/intersection-types.md:
```diff
@@ -11,18 +11,18 @@ The type `S & T` represents values that are of the type `S` and `T` at the same
 }
 trait Growable[T] {
-  def add(x: T): this.type
+  def add(t: T)
```
same here

odersky commented on docs/docs/reference/other-new-features/named-typeargs.md:
```diff
@@ -5,7 +5,7 @@ title: "Named Type Arguments"
 **Note:** This feature is implemented in Dotty, but is not expected to be part of Scala 3.0.
 
-Type arguments of methods can now be named, as well as by position. Example:
+Type arguments of methods can now be specified by name next to by position. Example:
```
I don't think _next to_ is correct here. "can now be specified by name as well as by position"?

robstoll commented:
@odersky I see my confusion, I added a sentence to implicit-resolution so that others aren't: https://github.com/lampepfl/dotty/pull/8974/files#diff-843899951c4ebf33685c569656144812R5
That's the only addition, otherwise I have fixed the doc according to your suggestions:

odersky commented on docs/docs/reference/changed-features/implicit-resolution.md:
```diff
@@ -2,7 +2,7 @@
 title: "Changes in Implicit Resolution"
 ---
-
+This page describes changes to the implicit resolution of old-style `implicit`s in Dotty.
```
```suggestion
This page describes changes to the implicit resolution that apply both to the new `given`s and to the old-style `implicit`s in Dotty.
```

odersky commented:
@robstoll Great. Yes, that sentence you added is important. I just made a suggestion to clarify it further. Otherwise all is good.
